pkg/envinfo/envinfo_test.go

```diff
@@ -0,0 +1,307 @@
+	}{
+		{
+			name:      fmt.Sprintf("Case 1: %s to test", URL),
+			parameter: URL,
+			value:     testURL,
+			existingEnvInfo: EnvInfo{
+				componentSettings: ComponentSettings{},
+			},
+			expectError: false,
+		},
+		{
+			name:      fmt.Sprintf("Case 2: %s to test", invalidParam),
+			parameter: invalidParam,
+			value:     testURL,
+			existingEnvInfo: EnvInfo{
+				componentSettings: ComponentSettings{},
+			},
+			expectError: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			esi, err := NewEnvSpecificInfo("")
+			if err != nil {
+				t.Error(err)
+			}
+			esi.EnvInfo = tt.existingEnvInfo
+			err = esi.SetConfiguration(tt.parameter, tt.value)
+			if err == nil && tt.expectError {
+				t.Errorf("expected error for SetConfiguration with %s", tt.parameter)
+			} else if !tt.expectError {
+				if err != nil {
+					t.Error(err)
+				}
+
+				isSet := esi.IsSet(tt.parameter)
+
+				if !isSet {
+					t.Errorf("the '%v' is not set", tt.parameter)
+				}
+
+			}
+
+		})
+	}
+}
+
+func TestUnsetEnvInfo(t *testing.T) {
+	fs := filesystem.NewFakeFs()
+	tempEnvFile, err := fs.TempFile("", "odoenvinfo")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tempEnvFile.Close()
+	os.Setenv(envInfoEnvName, tempEnvFile.Name())
+	testURL := ConfigURL{Name: "testURL", ClusterHost: "1.2.3.4.nip.io", TLSSecret: "testTLSSecret"}
+	invalidParam := "invalidParameter"
+
+	tests := []struct {
+		name            string
+		parameter       string
+		existingEnvInfo EnvInfo
+		expectError     bool
+	}{
+		{
+			name:      fmt.Sprintf("Case 1: unset %s", URL),
+			parameter: URL,
+			existingEnvInfo: EnvInfo{
+				componentSettings: ComponentSettings{
+					URL: &[]ConfigURL{testURL},
+				},
+			},
+			expectError: false,
+		},
+		{
+			name:      fmt.Sprintf("Case 2: unset %s", invalidParam),
+			parameter: invalidParam,
+			existingEnvInfo: EnvInfo{
+				componentSettings: ComponentSettings{
+					URL: &[]ConfigURL{testURL},
+				},
+			},
+			expectError: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			esi, err := NewEnvSpecificInfo("")
+			if err != nil {
+				t.Error(err)
+			}
+			esi.EnvInfo = tt.existingEnvInfo
+			err = esi.DeleteConfiguration(tt.parameter)
+			if err == nil && tt.expectError {
+				t.Errorf("expected error for DeleteConfiguration with %s", tt.parameter)
+			} else if !tt.expectError {
+				if err != nil {
+					t.Error(err)
+				}
+				isSet := esi.IsSet(tt.parameter)
+				if isSet {
+					t.Errorf("the '%v' is not set to nil", tt.parameter)
+				}
+			}
+
+		})
+	}
+}
+
+func TestDeleteURLFromMultipleURLs(t *testing.T) {
+	tempEnvFile, err := ioutil.TempFile("", "odoenvinfo")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tempEnvFile.Close()
+	os.Setenv(envInfoEnvName, tempEnvFile.Name())
+	deleteParam := "testURL1"
+	testURL1 := ConfigURL{Name: deleteParam, ClusterHost: "1.2.3.4.nip.io", TLSSecret: "testTLSSecret"}
```

Comment:
please convert this test into the current table format and add a case when a single URL exists and deletion is triggered on it.

Reply:
updated